unittests/framework/utils/test_iem.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2001-2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+
+import time
+import unittest
+
+from cortx.utils.iem_framework import EventMessage
+from cortx.utils.process import SimpleProcess
+from framework.utils.iem import Iem
+
+
+class TestIEM(unittest.TestCase):
+
+    def setUp(self):
+        self.module = None
+        self.event_code = None
+        self.EVENT_CODE = {
+        "IPMITOOL_ERROR" : ["0050010001", "ipmitool"],
+        "IPMITOOL_AVAILABLE" : ["0050010002", "ipmitool"]
+        }
+        EventMessage.init(component='sspl', source='S')
+        self.iem = Iem()
+        # Check existing iem alert present in MessageBus
+        EventMessage.subscribe(component='sspl')
+        while True:
+            msg = EventMessage.receive()
+            if msg is None:
+                break
+
+    def test_iem_fault_alert_receive(self):
+        """Test iem 'ipmitool' fault alert receive."""
+        print("### TestCase: test_iem_fault_alert_receive\n")
+        self.iem.iem_fault("IPMITOOL_ERROR")
+        time.sleep(10)
```

Comment:
Why are we waiting for 10 seconds? Are we using the Kafka service in backend here? If so, we are not supposed to do that. Unittest should not be relying on any external running services. Either you can mock the service or move this test to sanity test suite.

Reply:
Addressed @thavanathan.

As discussed, any test having dependancy on external service(for eg:- kafka) should be added in sspl_test framework not unittest.
so, added a test to test iem alerts in alerts.pln